### PR TITLE
Apply type checking on monster type

### PR DESCRIPTION
**Pull request by AJenbo**
Follow up to @obligaron 's comments on #4601

### CODE REVIEW

File: Source/missiles.cpp [hunk 1]
```diff
@@ -2101,19 +2101,17 @@ void InitMissileAnimationFromMonster(Missile &mis, Direction midir, const Monste
...
+	Monster &monster = Monsters[missile._misource];
 
 	MonsterGraphic graphic = MonsterGraphic::Special;
-	if (monster.MType->mtype < MT_HORNED || monster.MType->mtype > MT_OBLORD) {
-		if (monster.MType->mtype < MT_NSNAKE || monster.MType->mtype > MT_GSNAKE) {
-			graphic = MonsterGraphic::Walk;
-		} else {
-			graphic = MonsterGraphic::Attack;
-		}
+	if (IsAnyOf(monster.MType->mtype, MT_NSNAKE, MT_RSNAKE, MT_BSNAKE, MT_GSNAKE)) {
```
Comment (obligaron) [inline Source/missiles.cpp:2107]:
```suggestion
	if (IsAnyOf(monster.MType->mtype, MT_NSNAKE, MT_RSNAKE, MT_BSNAKE, MT_GSNAKE)) {
		graphic = MonsterGraphic::Attack;
	} else if (IsNoneOf(monster.MType->mtype, MT_HORNED, MT_MUDRUN, MT_FROSTC, MT_OBLORD)) {
		graphic = MonsterGraphic::Walk;
```

The result is swapped or?
Only snakes use the attack animation. All other types use Walk.

We also could remove the negation:

```cpp
	MonsterGraphic graphic = MonsterGraphic::Walk;
	if (IsAnyOf(monster.MType->mtype, MT_HORNED, MT_MUDRUN, MT_FROSTC, MT_OBLORD)) {
		graphic = MonsterGraphic::Special;
	}
	else if (IsAnyOf(monster.MType->mtype, MT_NSNAKE, MT_RSNAKE, MT_BSNAKE, MT_GSNAKE)) {
		graphic = MonsterGraphic::Attack;
	}
```

**Review by obligaron — CHANGES_REQUESTED** [review]
Found two copy&past errors.
Else it looks much nicer and allows searching for a monster type in the code. 👍

File: Source/missiles.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/missiles.cpp:2107]: Good catch, also much nicer :)

**Review by obligaron — APPROVED** [review]
lgtm 🙂

**Pull request merged**